main.py

```diff
@@ -522,34 +605,38 @@ def makeBatch(replay_buffer):
 			latents = actions[:,:,:] # "cheat" to generate structure.
 			
 			model.zero_grad()
-			wp, ap, rp, a1, a2, w1, w2 = model.forward(board, latents, uu)
+			# Input the initial board state, the actions (embedded as latents) that cause the
+			# final board state, and the total number of epochs run
+			final_board_pred, actions_pred, rewards_pred, a1, a2, w1, w2 = model.forward(board, latents, total_epochs_run)
 			
+			# Loss objective: predict the final board state, actions latents that cause the final board state, and 
+			# the rewards that occur due to the actions. Note that action latents are given as input.
 			# keep the batch dim
-			loss = th.sum((new_board - wp)**2, (1,2))*0.15 + \
-						th.sum((actions - ap)**2, (1,2)) + \
-						(rewards[:,0,0] - rp[:,0,0])**2 * 4.0
+			loss = th.sum((new_board - final_board_pred)**2, (1,2))*0.15 + \
```

Comment:
'final' implies a terminal state -- which it usually is not; if we're coming up with better names, how about new_board_pred if the input is board and the target is new_board?

Reply:
sg updated to new_board